File: ainote-product/repos/api/note/src/main/java/com/newtech/note/entity/dto/noteModules/AIPictureModule.java
package com.newtech.note.entity.dto.noteModules;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.newtech.note.common.enumeration.NoteModuleType;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;
import lombok.Setter;
import org.bson.codecs.pojo.annotations.BsonDiscriminator;

@Setter
@Getter
@BsonDiscriminator(value = "NOTE_MODULE.AI_PICTURE")
@Schema(description = "AI图片模块")
public class AIPictureModule implements NoteModule {
    @Schema(description = "推荐商品模块ID")
    private String moduleId;

    @Schema(description = "模块类型")
    private NoteModuleType noteModuleType = NoteModuleType.AI_PICTURE;

    @Schema(description = "图片URL")
    private String imageUrl;

    @Schema(description = "标题")
    private String title;

    @Schema(description = "描述")
    private String description;

    @Schema(description = "分词结果")
    @JsonIgnore
    private String segmentation;
}

File: ainote-product/repos/ainote_app/lib/app/data/models/note_module/ai_picture_module.dart
import 'note_module_payload.dart';

class AIPictureModule extends NoteModulePayload {
  String? moduleId;
  String? title;
  String? description;
  String? imageUrl;

  AIPictureModule({
    this.moduleId,
    this.title,
    this.description,
    this.imageUrl,
  });

  factory AIPictureModule.fromJson(Map<String, dynamic> json) {
    return AIPictureModule(
      moduleId: json['moduleId'],
      title: json['title'],
      description: json['description'],
      imageUrl: json['imageUrl'],
    );
  }

  @override
  String get noteModuleType =>
      NoteModuleType.AI_PICTURE.toString().split('.').last;

  @override
  Map<String, dynamic> toJson() {
    return {
      'moduleId': moduleId,
      'noteModuleType': noteModuleType,
      'title': title,
      'description': description,
      'imageUrl': imageUrl,
    };
  }
}
